File: client/android/src/com/wireguard/android/backend/GoBackend.java
package com.wireguard.android.backend;


public final class GoBackend {
    private static final String TAG = "WireGuard/GoBackend";

    public static native String wgGetConfig(int handle);

    public static native int wgGetSocketV4(int handle);

    public static native int wgGetSocketV6(int handle);

    public static native void wgTurnOff(int handle);

    public static native int wgTurnOn(String ifName, int tunFd, String settings);

    public static native String wgVersion();

}

File: client/android/wireguard/src/main/kotlin/com/wireguard/android/backend/GoBackend.kt
package com.wireguard.android.backend

// TODO: Refactor Amnezia wireguard project by changing the JNI method names
//  to move this object to 'org.amnezia.vpn.protocol.wireguard.backend' package
object GoBackend {
    external fun wgGetConfig(handle: Int): String?
    external fun wgGetSocketV4(handle: Int): Int
    external fun wgGetSocketV6(handle: Int): Int
    external fun wgTurnOff(handle: Int)
    external fun wgTurnOn(ifName: String, tunFd: Int, settings: String): Int
    external fun wgVersion(): String
}
